src/Middleware/StaticFiles/src/StaticFileContext.cs

```diff
@@ -289,9 +292,8 @@ private static PreconditionState GetMaxPreconditionState(params PreconditionStat
 
     public Task SendStatusAsync(int statusCode)
     {
-        ApplyResponseHeaders(statusCode);
+        ApplyResponseHeadersAsync(statusCode);
 
-        _logger.Handled(statusCode, SubPath);
         return Task.CompletedTask;
```

Comment:
I think @Tratcher meant something like this in https://github.com/dotnet/aspnetcore/pull/45062#discussion_r1022179874:
```c#
_logger.Handled(statusCode, SubPath);
return ApplyResponseHeadersAsync(statusCode);
```